### DISCUSSION TITLE
Can I use Comodo SSL instead of Lets Encrypt?

### DISCUSSION THREAD

**Discussion by chrisferrantelli** [mastodon/mastodon#20087] [Q&A] [answered]
My domain comes with Comodo SSL, can I just use that instead of Lets Encrypt? Or does Mastodon only work with Lets Encrypt?

If I can use Comodo SSL how would I go about doing the nginx step with it?

**Comment by MrGeneration** [comment, marked as answer]
Yes of course.
You basically skip the letsencrypt step and put your ssl certificates onto your server in a directory your webserver can read.

These are the related lines:
https://github.com/mastodon/mastodon/blob/main/dist/nginx.conf#L36-L38

Possible directories e.g. are `/etc/ssl/*` or creating a `ssl` directory within `/etc/nginx`.

**Reply by chrisferrantelli** [reply]
Can it be the same format as this? `/etc/letsencrypt/live/example.com/fullchain.pem`? So like `/etc/ssh/live/example.com/fullchain.pem`?

**Reply by MrGeneration** [reply]
If that's what you want, of course!

**Reply by chrisferrantelli** [reply]
Actually I'm curious, if I activated it on my domain do I still need the keys?

**Reply by MrGeneration** [reply]
You always need the certificate together with it's private key if you want your webserver to ship it.

**Reply by chrisferrantelli** [reply]
Got, and sorry for all the questions. Really want to get this up and running and make sure I understand it. Would you know if the certs have to be `*.pem`? Because the way my domain host has it configured the ones I have are `fullchain.crt` for the full chain and `privatekey.p12`

**Reply by MrGeneration** [reply]
You'll want the private key in pem format. p12 is a container format which is not helping in that scenario.
